File: tests/gen_reactnative/test_interface/expected/ios/Setter.swift
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

import Foundation

@objc(Setter)
class RNSetter: NSObject {

    let nativeInstance = Setter()

    @objc(setBoolean:withResolver:withRejecter:)
    func setBoolean(value: Bool, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        nativeInstance.setBoolean(value: value)
        resolve(nil)
    }

    @objc(setInt:withResolver:withRejecter:)
    func setInt(value: NSInteger, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        nativeInstance.setInt(value: value)
        resolve(nil)
    }

    @objc(setFloat:withResolver:withRejecter:)
    func setFloat(value: Float, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        nativeInstance.setFloat(value: value)
        resolve(nil)
    }

    @objc(setString:withResolver:withRejecter:)
    func setString(value: NSString, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        nativeInstance.setString(value: value)
        resolve(nil)
    }

    @objc(setMap:withResolver:withRejecter:)
    func setMap(value: NSDictionary, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        nativeInstance.setMap(value: value)
        resolve(nil)
    }

    @objc(setList:withResolver:withRejecter:)
    func setList(value: NSArray, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        nativeInstance.setList(value: value)
        resolve(nil)
    }

}

File: tests/gen_android/test_interface/expected/dd-sdk-android/src/main/java/com/datadog/android/bridge/Setter.kt
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

package com.datadog.android.bridge

import android.content.Context

/**
 * An interface to test setting types
 */
class Setter(context: Context) {

    private val appContext = context.applicationContext

    /**
     * Empty method, boolean param, returns void
     */
    fun setBoolean(value: Boolean): Unit {
        TODO()
    }

    /**
     * Empty method, int param, returns void
     */
    fun setInt(value: Int): Unit {
        TODO()
    }

    /**
     * Empty method, float param, returns void
     */
    fun setFloat(value: Float): Unit {
        TODO()
    }

    /**
     * Empty method, string param, returns void
     */
    fun setString(value: String): Unit {
        TODO()
    }

    /**
     * Empty method, map param, returns void
     */
    fun setMap(value: Map<String, Any?>): Unit {
        TODO()
    }

    /**
     * Empty method, list param, returns void
     */
    fun setList(value: List<Any?>): Unit {
        TODO()
    }

}
